### ISSUE TITLE
ThemeData.accentColor has insufficient contrast for text

### ISSUE THREAD

**Issue by jonahwilliams** [flutter/flutter#19946]
For example, ButtonTextTheme.accent is the default value for ButtonTheme.bar, which in turn impacts the DatePicker, TimePicker, and Dialogs.

By default, the ButtonTextTheme.accent will use the theme's accent color which is currently the 500 shade, whereas the next color with sufficient contrast is the 600 shade (primary).

If the accent color is going to remain the 500 shade, then we should audit all usage of it and remove it from any derivative themes where it would create a contrast issue. This would involved deprecating & removing ButtonTextTheme.accent, since it would be the same as ButtonTextTheme.primary.

cc @willlarche @HansMuller
